src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,124 @@
+import { EnvironmentType } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService } from './types';
+import { IKernel } from '../../../kernels/types';
+import { executeSilently } from '../../../kernels/helpers';
+
+export const minimumSupportedPandaVersion = '0.20.0';
+export const getVersionOfPandasCommand =
+    'import pandas as _VSCODE_pandas;print(_VSCODE_pandas.__version__);del _VSCODE_pandas';
+
+function isVersionOfPandaSupported(version: SemVer) {
+    return version.compare(minimumSupportedPandaVersion) > 0;
+}
+
+/**
+ * Responsible for managing dependencies of a Data Viewer.
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    constructor(
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IsCodeSpace) private isCodeSpace: boolean
+    ) {}
+
+    private packaging(kernel: IKernel): 'pip' | 'conda' {
+        const envType = kernel.kernelConnectionMetadata.interpreter?.envType;
+        const isConda = envType === EnvironmentType.Conda;
+        return isConda ? 'conda' : 'pip';
+    }
+
+    public async checkAndInstallMissingDependenciesOnEnvironment(): Promise<void> {
+        throw new Error('Not implemented');
+    }
+
+    public async checkAndInstallMissingDependenciesOnKernel(kernel: IKernel): Promise<void> {
+        const pandasVersion = await this.getVersionOfPandas(kernel);
+
+        if (pandasVersion) {
+            if (isVersionOfPandaSupported(pandasVersion)) {
+                return;
+            }
+            sendTelemetryEvent(Telemetry.PandasTooOld);
+            // Warn user that we cannot start because pandas is too old.
+            const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
+            throw new Error(DataScience.pandasTooOldForViewingFormat().format(versionStr));
+        }
+
+        sendTelemetryEvent(Telemetry.PandasNotInstalled);
+        await this.installMissingDependencies(kernel);
+    }
+
+    private async installMissingDependencies(kernel: IKernel): Promise<void> {
+        if (!kernel.session) {
+            sendTelemetryEvent(Telemetry.NoActiveKernelSession);
+            throw new Error(DataScience.noActiveKernelSession());
+        }
+
+        sendTelemetryEvent(Telemetry.PythonModuleInstall, undefined, {
+            action: 'displayed',
+            moduleName: ProductNames.get(Product.pandas)!
+        });
+
+        const selection = this.isCodeSpace
+            ? Common.install()
+            : await this.applicationShell.showErrorMessage(
+                  DataScience.pandasRequiredForViewing(),
+                  { modal: true },
+                  Common.install()
+              );
+
+        const command = `${this.packaging(kernel)} install pandas`;
```

Comment:
If we are running in the kernel I would think we might want to use the pip / conda commands. Pretty sure they are safer:
https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-pip
Can have issues otherwise:
https://jakevdp.github.io/blog/2017/12/05/installing-python-packages-from-jupyter/

Reply:
Done!